ilc/client/setupErrorHandlers.js

```diff
@@ -1,7 +1,52 @@
+import * as singleSpa from 'single-spa';
+import * as uuidv4 from 'uuid/v4';
+
 const System = window.System;
 const newrelic = window.newrelic;
 
-export default function () {
+const KIND_OF_PRIMARY_FRAGMENT = 'primary';
+const KIND_OF_ESSENTIAL_FRAGMENT = 'essential';
+
+const selectFragmentKind = (registryConf, path, appName, slotName) => {
+    const appKind = registryConf.apps[appName].kind;
+    const slotKind = path.slots[slotName] && path.slots[slotName].kind;
+
+    return slotKind || appKind;
+}
+
+const isEssentialOrPrimaryFragment = (fragmentKind) => [KIND_OF_PRIMARY_FRAGMENT, KIND_OF_ESSENTIAL_FRAGMENT].includes(fragmentKind);
+
+export default function ({
+    registryConf,
+    getCurrentPath,
+}) {
+    singleSpa.addErrorHandler((err) => {
+        const errorId = uuidv4();
+        const [appName, slotName] = err.appOrParcelName.split('__at__');
+
+        const currentPath = getCurrentPath();
+        const fragmentKind = selectFragmentKind(registryConf, currentPath, `@portal/${appName}`, slotName);
+
+        if (isEssentialOrPrimaryFragment(fragmentKind)) {
+            fetch(`/_ilc/page/500/${errorId}`)
+                .then((res) => {
+                    if (!res.ok) {
+                        return Promise.reject(new Error('Something went wrong!'));
+                    }
+
+                    return res.text()
+                })
+                .then((text) => {
+                    document.querySelector('html').innerHTML = text;
+                })
+                .catch((err) => {
+                    alert(err.message);
```

Comment:
1. Don't show random `err.message` to user
2. Pls log this error with enough context info